### PR TITLE
Support CSAR packaging and Tosca metadata

### LINKED ISSUE #53: Support CSAR packaging and Tosca metadata directory
Support packaging projects as a CSAR (Zip) instead of a TGZ.
Also allow a TOSCA-Metadata directory to be included in the package. 

Changes have been trialled on the includeArtifacts branch.

### PR DESCRIPTION
**Pull request by dvaccarosenna**
Closes #53

### CODE REVIEW

File: lmctl/project/processes/package.py [hunk 1]
```diff
@@ -38,23 +43,35 @@ def execute(self):
...
 
+    def __build_package(self, add_method, pkg_tree, compiled_content_path, pkg_meta_file_path):
+        rootlen = len(compiled_content_path) + 1
+        for root, dirs, filelist in os.walk(compiled_content_path):
+            for file_name in filelist:
+                full_path = os.path.join(root, file_name)
+                file_size = os.path.getsize(full_path)
+                arcname = full_path[rootlen:]
+                if file_size > 100000000:
+                    # For big files let people know. TODO: make this more generic, so we can report long running tasks as events
```
Comment (chowarth123) [inline lmctl/project/processes/package.py:70]: is this TODO just considered outside of the scope of this issue?

File: docs/projects/intro.md [hunk 2]
```diff
@@ -35,6 +35,21 @@ The following table details the relevance of each item in the project tree:
 
 The remaining structure of the project, and each subproject, depends on the type of service under development.
 
+## TOSCA Additions
+
+In addition to the normal project structure, you may also include a `TOSCA-Metadata` directory to be included in the final package. This directory is expected to have a single `TOSCA.meta` file in.
```
Comment (chowarth123) [inline docs/projects/intro.md:40]: why do we have to have a directory with just a single file in it? will this be added to in future?

File: lmctl/project/processes/package.py (hunk 1 above)
Comment (dvaccarosenna) [reply lmctl/project/processes/package.py:70]: Yeah out of scope. Just a reminder as to why we log anything. This was actually already in the code and just got moved.

File: docs/projects/intro.md (hunk 2 above)
Comment (dvaccarosenna) [reply docs/projects/intro.md:40]: No idea. It's part of the Tosca spec, so we'd only add to it if they add anything to it.

**Pull request merged**